scripts/bin/gpu-2404-provider-wrapper.in

```diff
@@ -45,6 +45,10 @@ if [ -d "/var/lib/snapd/lib/gl32/vdpau" ]; then
   LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/var/lib/snapd/lib/gl32/vdpau
 fi
 
+if [ -d "/var/lib/snapd/lib/gl/gbm" ]; then
+  GBM_BACKENDS_PATH=/var/lib/snapd/lib/gl/gbm
```

Comment:
I have one doubt with this. If we somehow end up shipping gbm backends within the `mesa-2404` snap, will the default path still get searched?

I suppose it's moot, since there's only the Nvidia alternate backend that we can think of right now?

Reply:
Just to answer my question, though - no, it won't:

https://gitlab.freedesktop.org/mesa/mesa/-/blob/21ce1ca846698527093b0f8b95a57c550f984ab1/src/loader/loader.c#L525-527
```
 * \param search_path_vars - NULL-terminated list of env vars that can be used
 * \param default_search_path - a colon-separted list of directories used if
 * search_path_vars is NULL or none of the vars are set in the environment.
```

Reply:
Right. But the default search path will be useless because it points to `/usr/lib/x86_64-linux-gnu/gbm` and not `$SNAP/gpu-2404/...`

In that case you would just extend the `$GBM_BACKENDS_PATH` variable with a colon-separated list of directories, like you currently do for `$LIBGL_DRIVERS_PATH`

----
unrelated but on the same topic: see this change for mesa 24.2, does it still work in snap confinement? https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/29771

Reply:
> In that case you would just extend the `$GBM_BACKENDS_PATH` variable with a colon-separated list of directories, like you currently do for `$LIBGL_DRIVERS_PATH`

Oh does it support colon-separation? Didn't find that documented anywhere... We'll just have to set it to `$SNAP/...` if we ever start shipping something there.

> unrelated but on the same topic: see this change for mesa 24.2, does it still work in snap confinement? https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/29771

AIUI that will then rely on `LD_LIBRARY_PATH`, so that will work fine.

Reply:
Yea I had to dig into the implementation but indeed it supports multiple paths with colon separation!